### PR TITLE
Support multiple keys splitted by '||'

### PR DESCRIPTION
**Pull request by winebarrel**
Add support for multiple keys:

```ruby
ENV['DB_USER'] = nil
ENV['DB_USER2'] = "db_user2"

# "#{DB_USER || DB_USER2}@..." => "db_user2@..."
```

### CODE REVIEW

File: lib/secret_env.rb [hunk 1]
```diff
@@ -37,20 +37,24 @@ def value
       scanner = StringScanner.new(@raw_value)
       parts = []
       while part = scanner.scan_until(/#\{(.*?)\}/)
-        secret_key = scanner.matched[2..-2] # Extract "secret" from "\#{secret}"
+        secret_keys = scanner.matched[2..-2] # Extract "secret" from "\#{secret}"
 
+        secret = nil
+        secret_keys.split(/\s*\|\|\s*/).each do |secret_key|
```
Comment (adorechic) [inline lib/secret_env.rb]: I think this splitting may contain unnecessary spaces

```ruby
irb(main):005:0> " db_password || db_password2".split(/\s*\|\|\s*/)
=> [" db_password", "db_password2"]
```

How about this?

```ruby
irb(main):003:0> "db_password || db_password2".split("||").map(&:strip)
=> ["db_password", "db_password2"]
```

**Review by adorechic — CHANGES_REQUESTED** [review]

File: lib/secret_env.rb (hunk 1 above)
Comment (winebarrel) [reply lib/secret_env.rb]: > How about this?

Looks good. :-)
FIxed d529ad3

**Review by adorechic — APPROVED** [review]

**Comment by adorechic** [comment]
https://rubygems.org/gems/secret_env/versions/0.4.0

**Comment by winebarrel** [comment]
Thanks!

**Pull request merged**